spec/features/admin/reports_spec.rb

```diff
@@ -33,7 +34,46 @@
     User.student.create
     User.instructor.create
     # run report
-    visit '/admin/reports'
+    visit admin_reports_path
     expect(page).to have_content("Student accounts created in the past week: 1")
   end
-end
\ No newline at end of file
+
+  example 'How many people start the account creation process versus how many finish it?' do
+    years = (2016..2020).to_a
+    years.each_with_index do |year, index|
+        Timecop.freeze(Time.local(year) + 1.day) do
+          (index * 1).times { FactoryBot.create(:pre_auth_state) }
+          (index * 2).times { User.create(state: User::ACTIVATED) }
+        end
+    end
+
+    visit admin_reports_path
+
+    expect(page).to have_content('2018: 2 started / 4 finished = 50.0%')
+    expect(page).to have_content('2019: 3 started / 6 finished = 50.0%')
+    expect(page).to have_content('2020: 4 started / 9 finished = 44.44%') # 8 finished plus the admin
+  end
+
+  context "What's the average time someone spends signing up (started the process but did not finish)?" do
```

Comment:
view tests are so much slower than isolated unit tests (mentioned above).  way better to isolate unit test and have minimal view testing.  view testing is usually best as golden path.